added changes as required

designanderror opened the pull request:
- Added hashing logic
- Modified output file to include hashed values
- Added logic to get certs from 2017
- Remove the filtering for unexpired certs
- added splitting of output files -> 10 million as max entry per file

<!---
Describe your changes in detail here.
If this fixes an issue, please write "Fixes #123", substituting the issue number.
-->

### Checklist

<!---
Go over all the following points, and put an `x` in all the boxes that apply.
Feel free to not tick any boxes that don't apply to this PR (e.g. refactoring may not need a CHANGELOG update).
If you're unsure about any of these, don't hesitate to ask. We're here to help!
-->

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

designanderror committed:
added changes as required

- Added hashing logic
- Modified output file to include hashed values
- Added logic to get certs from 2017
- Remove the filtering for unexpired certs
- added splitting of output files -> 10 million as max entry per file

steurer commented on main/main.go:
```diff
@@ -42,11 +48,11 @@ func main() {
-		panic(err)
-	}
-	defer closeFunc()
+	// out, closeFunc, err := getFileWriter(*out, zstd.SpeedBetterCompression)
```
remove commented code

steurer commented on main/main.go:
```diff
@@ -76,25 +82,72 @@ func main() {
 
+	// Define a variable to keep track of the number of entries in the current output file
+	entriesWritten := 0
+	maxEntriesPerFile := 10_000_000 // 10 million entries
```
make configurable

steurer commented on main/main.go:
```diff
@@ -76,25 +82,72 @@ func main() {
+	if err != nil {
+		panic(err)
+	}
+	defer closeFunc()
```
can remove

steurer commented on main/main.go:
```diff
@@ -76,25 +82,72 @@ func main() {
 	}()
 
-	now := time.Now()
+	// nameChan := make(chan resultEntry, 1000)
```
remove

steurer commented on main/main.go:
```diff
@@ -117,13 +170,17 @@ func main() {
+				hash, names := getDomainNames(parsedEntry)
+				nameChan <- resultEntry{
+					hash:      hash,
+					name:      strings.Join(names, ","), // Join domain names with a comma
```
semicolon or use names at the end

steurer commented on main/main.go:
```diff
@@ -144,14 +201,19 @@ func main() {
+				hash, names := getDomainNames(parsedEntry)
+				nameChan <- resultEntry{
+					hash:      hash,
+					name:      strings.Join(names, ","), // Join domain names with a comma
```
see above

steurer commented on main/main.go:
```diff
@@ -165,31 +227,36 @@ func main() {
-	return names
+
+	// Create a hash over the domain names
+	hashBytes := sha256.Sum256([]byte(strings.Join(names, ",")))
```
1. Sort names before hashing
2. Normalize domain names here (use a lib or standard library)

steurer commented on main/main.go.txt:
```diff
@@ -0,0 +1,215 @@
+			if *noPrecert {
+				return nil
+			}
+
```
delete this file

designanderror committed:
Update main.go

- converted max entries as configurable (maxen)
- added sorting and normalizing of domain names

designanderror committed:
Delete main/main.go.txt

designanderror committed:
Delete scanner/scanlog/scanlog.exe

designanderror committed:
Delete scanner/scanlog/scanlog.go.txt

designanderror committed:
Delete scanner/scanlog/output.txt

designanderror committed:
Update main.go

designanderror committed:
Update main.go

removed hashing

designanderror committed:
added leafTimestamp

designanderror committed:
Merge branch 'master' of https://github.com/designanderror/certificate-transparency-go

designanderror committed:
added cname

Output Final Structure:
Index
CName
DNS Names
IsPrecert
validFrom
validTo
leafTime

designanderror committed:
updated cname for precert

Output Final Structure:
Index
CName
DNS Names
IsPrecert
validFrom
validTo
leafTime

designanderror committed:
processing download file

designanderror committed:
Update play.go

fixed the return hash value issue

designanderror committed:
fixed error

fixed the "bufio.Scanner: token too long" error

designanderror committed:
fixed token too long error

designanderror committed:
final data process

designanderror committed:
sorting files

designanderror committed:
solved the overwrite issue

designanderror committed:
Merge branch 'steurer:master-simran' into master
